compiler/src/main/scala/org/scalajs/core/compiler/PrepJSExports.scala

```diff
@@ -368,6 +368,15 @@ trait PrepJSExports[G <: Global with Singleton] { this: PrepJSInterop[G] =>
                 "You may not export a getter or a setter as a named export")
           }
 
+          def isUnnamedExportInsideExportAll = {
+            isMember && !hasExplicitName && annot.symbol == JSExportAnnotation &&
+              unitAnnots.exists(_.symbol == JSExportAllAnnotation)
+          }
+          if (isUnnamedExportInsideExportAll) {
+            reporter.warning(sym.pos, "Found useless @JSExport inside @JSExportTopLevel."
+              + """ Maybe @JSExport("someName") ?""")
+          }
+
```

Comment:
This would also be rendered unnecessary if we remove the special case below (IIUC).

Reply:
Simplified in https://github.com/scala-js/scala-js/pull/3633/commits/b0f391d38547cea9fc4372e11767a48dbfd15275